ql/experimental/callablebonds/discretizedcallablefixedratebond.cpp

```diff
@@ -73,10 +73,23 @@ namespace QuantLib {
                     couponAdjustments_[j] = CouponAdjustment::pre;
 
                     /* We snapped the callabilityTime so we need to take into account the missing
-                     * discount factor. */
-                    auto discountTillCallDate = termStructure->discount(callabilityDate);
-                    auto discountTillCouponDate = termStructure->discount(couponDate);
-                    adjustedCallabilityPrices_[i] *= discountTillCallDate / discountTillCouponDate;
+                     * discount factor including any possible spread e.g. set in the OAS
+                     * calculation. */
+                    auto spread  = arguments_.spread;
+                    auto calcDiscountFactorInclSpread = [&termStructure, spread](Date date) {
+                        auto time = termStructure->dayCounter().yearFraction(
+                            Settings::instance().evaluationDate(), date);
```

Comment:
I think this should be calculated from the reference date of the curve (that's how the existing calls to `discount` worked) in which case it can be simplified to
```
                        auto time = termStructure->timeFromReference(date);
```

Reply:
Sure, I wasn't aware of this feature!